File: backend/app/models/task.py
from typing import Optional
from pydantic import Field, ConfigDict

from .base import BaseIdentifiedModel
from .enums import TaskType, TaskStatus


class Task(BaseIdentifiedModel):
    model_config = ConfigDict(use_enum_values=True)
    
    type: TaskType = Field(..., description="任务类型")
    status: TaskStatus = Field(default=TaskStatus.PENDING, description="任务状态")
    progress: float = Field(default=0.0, ge=0.0, le=1.0, description="进度 (0.0-1.0)")
    current_step: int = Field(default=0, ge=0, description="当前步骤")
    total_steps: int = Field(default=0, ge=0, description="总步骤数")
    error_message: Optional[str] = Field(None, description="错误信息")

File: frontend/src/types/task.ts
// 任务相关类型定义

export interface Task {
  id: string;
  type: TaskType;
  status: TaskStatus;
  progress: number; // 0.0 - 1.0
  current_step: number;
  total_steps: number;
  error_message?: string;
}

export type TaskType = "script_generation" | "audio_generation" | "pdf_conversion";
export type TaskStatus = "pending" | "running" | "completed" | "failed";

// API 请求/响应类型
export interface GetTaskResponse {
  finished: boolean;
  task: Task;
}

export interface CancelTaskResponse {
  message: string;
  task_id: string;
}

// 任务进度监控相关类型
export interface TaskProgress {
  taskId: string;
  progress: number;
  status: TaskStatus;
  currentStep: number;
  totalSteps: number;
  errorMessage?: string;
}

export interface TaskMonitor {
  startMonitoring: (taskId: string) => void;
  stopMonitoring: () => void;
  getProgress: () => TaskProgress | null;
  addListener: (listener: (progress: TaskProgress) => void) => void;
  removeListener: (listener: (progress: TaskProgress) => void) => void;
}